### ISSUE TITLE
Password prompt

### ISSUE THREAD

**Issue by notresponding2u** [ayushnix/tessen#7]
Hi, 

I find your tessen as a great substitute of rofi-pass, given the technical context I am using - Arch linux, Wayland, Sway and wofi. Please consider that I am newbie in Arch and have some tolerance - the cool kids don't let me hang out with them oherwise. 

What I am trying to do Is to bind tessen to a key bind, here is the code I have added in my ~/.config/sway/config:
`
bindsym Alt+p exec tessen
`

I used to pull that/similar trick in Ubuntu with rofi-pass and it worked great. I do **not** assume this is the correct way of doing.

The issue I have, is that when I run tessen with the key combination mentioned above, when it is the time to ask for a password to use the gpg key, which encrypts my secrets, no password prompt shows up. However, if I run tessen in the terminal (Alacrity with fish shell), I get password prompt, and after that I can use tessen through the shortcut combo while it is still authorized without any problems.

In short, how should I configure my system to get gui password prompt when I use tessen with a shortcut?

I know one of my colleagues is using rofi-pass in the same environment (Arch, Wayland and Sway), but he is on vacation and I want to make a good impression when he is back. :D I am not sure what trick he pulls to get the password prompt popup. Please help.

Sincerely

**Comment by ayushnix** [comment]
> when I run tessen with the key combination mentioned above, when it is the time to ask for a password to use the gpg key, which encrypts my secrets, no password prompt shows up

This is strange because I'm using a [similar method](https://github.com/ayushnix/dotfiles/blob/833678009b3f5bda81df65f2cff21500c9411462/private_dot_config/sway/config#L242) to launch tessen and I get the gpg pinentry menu which asks for my password without any issues.

I also installed wofi and used that as a backend for tessen and that worked fine as well.

Can you show the output of the following commands on your system?

```
pacman -Qi pinentry
grep '^pinentry-program' ~/.gnupg/gpg-agent.conf
```

You could also check the output of `journalctl -b -t gcr-prompter` if you're using `pinentry-gnome3` and see if there are any error messages.

**Comment by notresponding2u** [comment]
Thank you for your response.

The output:
`
~> pacman -Qi pinentry
Name            : pinentry
Version         : 1.2.0-1
Description     : Collection of simple PIN or passphrase entry dialogs which utilize the Assuan protocol
Architecture    : x86_64
URL             : https://gnupg.org/related_software/pinentry/
Licenses        : GPL
Groups          : None
Provides        : None
Depends On      : glibc  ncurses  libcap  libassuan  libsecret  glib2  libsecret-1.so=0-64
                  libcap.so=2-64  libncursesw.so=6-64  libassuan.so=0-64  libglib-2.0.so=0-64
Optional Deps   : gtk2: gtk2 backend
                  qt5-base: qt backend [installed]
                  gcr: gnome3 backend [installed]
Required By     : gnupg
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 563.46 KiB
Packager        : Levente Polyak <anthraxx@archlinux.org>
Build Date      : Sun Sep 19 20:53:53 2021
Install Date    : Thu Dec 30 09:18:07 2021
Install Reason  : Installed as a dependency for another package
Install Script  : No
Validated By    : Signature
`
`
~> grep '^pinentry-program' ~/.gnupg/gpg-agent.conf
grep: /home/lsd/.gnupg/gpg-agent.conf: No such file or directory
`

I guess the second output narrows it a bit, but I am still unsure what to do.

**Comment by ayushnix** [comment]
> > `grep '^pinentry-program' ~/.gnupg/gpg-agent.conf`
> > `grep: /home/lsd/.gnupg/gpg-agent.conf: No such file or directory`
> 
> I guess the second output narrows it a bit, but I am still unsure what to do.

You probably need to create `~/.gnupg/gpg-agent.conf` file and mention a pinentry program that should be used by default.

You should probably have these files on your system

```
[user@host ~]$ ls -lAh /usr/bin/pinentry-*
-rwxr-xr-x 1 root root  75K Sep 20 00:23 /usr/bin/pinentry-curses
-rwxr-xr-x 1 root root  67K Sep 20 00:23 /usr/bin/pinentry-emacs
-rwxr-xr-x 1 root root  87K Sep 20 00:23 /usr/bin/pinentry-gnome3
-rwxr-xr-x 1 root root  95K Sep 20 00:23 /usr/bin/pinentry-gtk-2
-rwxr-xr-x 1 root root 159K Sep 20 00:23 /usr/bin/pinentry-qt
-rwxr-xr-x 1 root root  67K Sep 20 00:23 /usr/bin/pinentry-tty
```

Pick one of the pinentry programs you'd like to use by default. You probably want either the gnome3 or the qt pinentry program. [Here's](https://github.com/ayushnix/dotfiles/blob/e958355c4eaafdd2c388b30ea5d311b7bed19b59/private_dot_gnupg/private_gpg-agent.conf) an example of how that file look like.

**Comment by notresponding2u** [comment]
Yes, I have the same pinentry programs, and I have set the same gpg-agent.conf. I took a look at yours earlier today and tried it out. 

I have also did `gpg-connect-agent updatestartuptty /bye >/dev/null` - still doesn't work. Not sure what is wrong. I should have probably mentioned that earlier, but I wasn't sure if the way I made it is correct. The location of the config is also correct `~/.gnupg/gpg-agent.conf`.

**Comment by notresponding2u** [comment]
Alright, I had a mistake in my gpg-agent.conf. I have corrected it and did `gpgconf --kill gpg-agent`. And it works as it should. Your comments definitely helped me get oriented in the situation.

Thank you for your time. Have a good new year.

**Comment by ayushnix** [comment]
Glad to know it worked out for you.

> Thank you for your time. Have a good new year.

You're welcome and you have a happy new year too.
